homeassistant/components/alarm_control_panel/group.py

```diff
@@ -0,0 +1,241 @@
+
+        self.hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_START, group_alarm_startup)
+
+    @asyncio.coroutine
+    def _async_send_message(self, service, **kwargs):
+        """Send message to all entities in the group."""
+        payload = {key: val for key, val in kwargs.items() if val}
+
+        tasks = []
+        for entity_id in self._entity_ids:
+            sending_payload = deepcopy(payload.copy())
+            sending_payload[ATTR_ENTITY_ID] = entity_id
+            tasks.append(self.hass.services.async_call(
+                DOMAIN, service, sending_payload))
+
+        if tasks:
+            yield from asyncio.wait(tasks, loop=self.hass.loop)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def code_format(self):
+        """The code format is customizable."""
+        return self._code_format
+
+    @asyncio.coroutine
+    def async_alarm_disarm(self, code=None):
+        """Send disarm command."""
+        yield from self._async_send_message(SERVICE_ALARM_DISARM,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_home(self, code=None):
+        """Send arm home command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_HOME,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_away(self, code=None):
+        """Send arm away command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_AWAY,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_night(self, code=None):
+        """Send arm night command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_NIGHT,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_trigger(self, code=None):
+        """Send alarm trigger command."""
+        yield from self._async_send_message(SERVICE_ALARM_TRIGGER,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_custom_bypass(self, code=None):
+        """Send arm custom bypass command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_CUSTOM_BYPASS,
+                                            code=code)
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        # _update_state always behaves as if a transition was in progress.
+        # But if the pre- and post-state match, we're not "pending"!
+        self._update_state()
+        if self._pre_state == self._post_state:
+            self._state = self._post_state
+        else:
+            self._state = STATE_ALARM_PENDING
+        return self._state
+
+    def _update_state(self):
```

Comment:
I would rename this to `async_update`.

Reply:
To avoid context switching between event loop and a thread it's better to have it as `async_update`.